docs/source/notes/randomness.rst

```diff
@@ -2,70 +2,73 @@ Reproducibility
 ===============
 
 Completely reproducible results are not guaranteed across PyTorch releases,
-individual commits or different platforms. Furthermore, results need not be
+individual commits, or different platforms. Furthermore, results need not be
 reproducible between CPU and GPU executions, even when using identical seeds.
 
-However, in order to make computations deterministic on your specific problem on
-one specific platform and PyTorch release, there are a couple of steps to take.
+However, there are some steps you can take to limit the number of sources of
+nondeterministic behavior for a specific platform, device, and PyTorch release.
 
-There are two pseudorandom number generators involved in PyTorch, which you will
-need to seed manually to make runs reproducible. Furthermore, you should ensure
-that all other libraries your code relies on and which use random numbers also
-use a fixed seed.
+PyTorch uses a pseudorandom number generator, which you will need to seed
+manually to make it act deterministically. Furthermore, you can configure
+PyTorch to avoid using nondeterministic algorithms for a subset of operations.
+Finally, if your code relies on other libraries that use random numbers, make
+sure to give them a fixed seed.
 
-PyTorch
-.......
+Random number generation
+........................
 You can use :meth:`torch.manual_seed()` to seed the RNG for all devices (both
 CPU and CUDA)::
 
     import torch
     torch.manual_seed(0)
 
+Deterministic algorithms
+........................
+Some PyTorch operations use nondeterministic algorithms by default. A subset
+of them can be configured to use a deterministic implementation or throw an
+error if a deterministic implementation is not available. Please check the
+documentation for :meth:`torch.set_deterministic()` for a full list of affected
+operations. If an operation does not act correctly according to the documentation,
+or if you need a deterministic implementation of an operation that does not have
+one, please submit an issue:
+`<https://github.com/pytorch/pytorch/issues?q=label:%22topic:%20determinism%22>`_
 
-There are some PyTorch functions that use CUDA functions that can be a source
-of nondeterminism. One class of such CUDA functions are atomic operations,
-in particular :attr:`atomicAdd`, which can lead to the order of additions being
-nondetermnistic. Because floating-point addition is not perfectly associative
-for floating-point operands, :attr:`atomicAdd` with floating-point operands can
-introduce different floating-point rounding errors on each evaluation, which
-introduces a source of nondeterministic variance (aka noise) in the result.
+.. code-block::
 
-PyTorch functions that use :attr:`atomicAdd` in the forward kernels include
-:meth:`torch.Tensor.index_add_`, :meth:`torch.Tensor.scatter_add_`,
-:meth:`torch.bincount`.
+    torch.set_deterministic(True)
```

Comment:
I feel like this code snippet can be cut and inlined in the preceding paragraph or elaborated upon, maybe by showing a nondeterministic function throwing an error?

Reply:
I've inlined it and also added an example of an error.